### PR TITLE
add frontend torch rand function

### LINKED ISSUE #9686: rand

### PR DESCRIPTION
**Pull request by jonnyroutley**
Close #9686 

Hi, 

Here are my initial attempts to add the torch rand function to the frontend API. My questions are:
- The [docs for rand](https://pytorch.org/docs/stable/generated/torch.rand.html#torch.rand) have some keyword arguments (like requires_grad, pin_memory) that I've included as arguments but haven't used (which I think I've seen happen elsewhere). Is this ok?
- As the function generates random numbers, I've set the test_values flag to False in the call to test_frontend_function. I was looking for a way to compare two different arrays using a manual seed but I don't think this is that straightforward. The other tests are running locally for me - are they sufficient?

Thanks!
Jonny

### CODE REVIEW

File: ivy_tests/test_ivy/test_frontends/test_torch/test_random_sampling.py [hunk 1]
```diff
@@ -139,3 +139,42 @@ def call():
...
+            num_positional_args=num_positional_args,
+            native_array_flags=native_array,
+            with_out=with_out,
+            frontend=frontend,
+            test_values=False,
+            fn_tree=fn_tree,
+            size=size
+        )
+
+    ret = call()
```
Comment (fspyridakos) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_random_sampling.py:180]: Add a shape and dtype check as other tests above

File: ivy/functional/frontends/torch/random_sampling.py [hunk 2]
```diff
@@ -54,3 +54,14 @@ def multinomial(input, num_samples, replacement=False, *, generator=None, out=No
 @to_ivy_arrays_and_back
 def poisson(input, generator=None):
     return ivy.poisson(input, shape=None)
+
+
+@to_ivy_arrays_and_back
+def rand(size, *, generator=None, out=None, dtype=None, layout=None, device=None,
```
Comment (fspyridakos) [inline ivy/functional/frontends/torch/random_sampling.py:60]: It's fine to ignore `pin_memory`, as with `requires_grad` I'm not sure and I'm waiting to hear back from the team member who's worked on gradients the most, will let you know when I hear back

File: ivy_tests/test_ivy/test_frontends/test_torch/test_random_sampling.py [hunk 3]
```diff
@@ -139,3 +139,42 @@ def call():
...
+@handle_frontend_test(
+    fn_tree="torch.rand",
+    dtype=helpers.get_dtypes("float", full=False),
+    size=helpers.get_shape(
+        min_num_dims=0,
+        max_num_dims=5,
+        min_dim_size=0,
+        max_dim_size=10,
+    ),
+    # seed=helpers.ints(min_value=0, max_value=100),
```
Comment (fspyridakos) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_random_sampling.py:153]: You can remove this seed generation

**Review by fspyridakos — CHANGES_REQUESTED** [review]
Hey thanks for your contribution! Good work so far, just a couple of comments on your test. As for the arguments, I'll get back to you on that soon

File: ivy_tests/test_ivy/test_frontends/test_torch/test_random_sampling.py (hunk 1 above)
Comment (jonnyroutley) [reply ivy_tests/test_ivy/test_frontends/test_torch/test_random_sampling.py:162]: tests added

**Comment by jonnyroutley** [comment]
Thank you for your comments! I've added the changes I think are required now - I've edited my test function to match the new handle_frontend_test(), hopefully I've done it correctly but please let me know

File: ivy/functional/frontends/torch/random_sampling.py (hunk 2 above)
Comment (jonnyroutley) [reply ivy/functional/frontends/torch/random_sampling.py:60]: marking as resolved for now but can reopen if requires_grad needs changing

**Comment by fspyridakos** [comment]
> Edit: I pulled latest commits from upstream (unifyai/ivy:master) and rebased my branch with these commits before adding my own stuff - this may not have been a good idea, but I think all the tests/checks initially passed before I submitted for re-review

Yes, now it seems like you're trying to make changes to 125 files, I would suggest you use `merge_with_upstream.sh` in the future, or just click the merge button in your branch on github. If you could somehow revert this it would be nice but it might jut be easier to make a new branch that's updated with `main` and open a new pr with your changes there.

**Comment by jonnyroutley** [comment]
gonna close this and create a new PR from a different branch - seems cleanest. New PR here: https://github.com/unifyai/ivy/pull/9808
